core/src/main/java/org/frankframework/jta/narayana/NarayanaConfigurationBean.java

```diff
@@ -98,34 +99,43 @@ public void afterPropertiesSet() throws ObjectStoreException {
 	}
 
 	/**
-	 * Tests if we can find a (valid) DataSource. Takes ApplicationServer specific jndiPrefixes into account.
-	 * If the DataSource is valid it will be returned with the {@link PoolingDataSourceJDBCAccess} prefix.
+	 * Find the datasource for the Narayano Object Store, specified by the property {@code transactionmanager.narayana.objectStoreDatasource}.
+	 * The datasource is first looked up in the {@code resources.yml} file, and if it cannot be found there it
+	 * will be looked up in the JNDI.
+	 * Since the datasource is used by Narayana internally, it cannot be an XA-Only datasource.
 	 */
-	private String getObjectStoreJndiName() throws ObjectStoreException {
+	private DataSource getObjectStoreDataSource() throws ObjectStoreException {
```

Comment:
Ik denk dat de oplossing prima is, volgens mij kan je in plaats van createBean ook een getBean doen.
Dan hoef je niet een nieuwe aan te maken en kan je de huidige DataSourceFactory gebruiken.

Mogelijk kan je de code opsplitsen? En dan individueel testen ofzo?

Reply:
`getBean` kon niet, dat had ik als eerste geprobeerd. Maar dan krijg je heel veel autowire-errors omdat er 2 beans zijn die DataSourceFactory implementeren, en dan weet Spring niet welke hij moet kiezen.
Ik had geen zin om uit te zoeken waar ik dat dan allemaal recht zou moeten gaan breien.

Code opsplitsen kan, maar ook de JNDI lookup voor de datasource wordt nu niet getest. Ik weet niet of het het testen makkelijker maakt als ik het opsplits.